File: Lab2/src/Model/MyList.java
package Model;


import java.util.ArrayList;
import java.util.List;

public class MyList<T> implements MyIList<T> {
    private List<T> list = new ArrayList<>();

    public MyList() {
    }

    public void add(T var) {
        this.list.add(var);
    }

    @Override
    public T fromIndex(int index) {
        return this.list.get(index);
    }


    @Override
    public String toString() {
        if (this.list.isEmpty())
            return "";
        StringBuilder res = new StringBuilder();
        for (T elem : this.list)
            res.append(elem).append("\n");

        return res.toString();
    }

    @Override
    public void clear() {
        this.list.clear();
    }

    @Override
    public int size() {
        return this.list.size();
    }

}

File: A7/ToyLanguage/Model/MyList.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ToyLanguage.Model
{
    public class MyList<T> : IList<T>
    {
        private List<T> list = new List<T>();
        public MyList()
        {
        }

        public void add(T var)
        {
            this.list.Add(var);
        }

        public void clear()
        {
            this.list.Clear();
        }

        public T fromIndex(int index)
        {
            return this.list[index];
        }

        public int size()
        {
            return this.list.Count();
        }

        public override string ToString()
        {
            String toStr = "";
            for (int i = 0; i < this.list.Count(); ++i)
            {
                toStr += this.list[i] + "\n";
            }
            return toStr;
        }
    }
}
